src/cargo/ops/resolve.rs

```diff
@@ -153,14 +160,37 @@ pub fn resolve_with_previous<'a>(registry: &mut PackageRegistry,
     //    still matches the locked version.
     if let Some(r) = previous {
         trace!("previous: {:?}", r);
-        for node in r.iter().filter(|p| keep(p, to_avoid, &to_avoid_sources)) {
+        for node in r.iter().filter(keep) {
             let deps = r.deps_not_replaced(node)
-                        .filter(|p| keep(p, to_avoid, &to_avoid_sources))
+                        .filter(keep)
                         .cloned().collect();
             registry.register_lock(node.clone(), deps);
         }
     }
 
+    for (url, augments) in ws.root_augment() {
+        let previous = match previous {
+            Some(r) => r,
+            None => {
+                registry.augment(url, augments)?;
+                continue
+            }
+        };
+        let augments = augments.iter().map(|dep| {
+            let unused = previous.unused_augments();
```

Comment:
Hm, what if `previous` contains some old augments, which are no longer present in the current `Cargo.toml`?